python/tank/util/version.py

```diff
@@ -115,6 +122,20 @@ def is_version_number(version):
         return False
 
 
+@contextlib.contextmanager
+def suppress_known_deprecation():
+    """
+    Imported function from setuptools.distutils module
+    """
+    with warnings.catch_warnings(record=True) as ctx:
+        warnings.filterwarnings(
+            action="default",
+            category=DeprecationWarning,
+            message="distutils Version classes are deprecated.",
```

Comment:
I am confused now. Should we not use this method then?

Reply:
You can find more information in the PR description. To sum up, distutils is now deprecated, and starting from Python 10 we will use the distutils copy that is shipped on `setuptools`.

Reply:
Ok but this is a temporary solution, right? At some point, we need to migrate to `packaging.version.parse`, no?
If so, do we have a plan for that (a Jira ticket)? And why not now (too big?)?

Reply:
Yes, it's too big. At first, I tried the migration to `packaging.version.parse`  but the existing code for tk-core, tk-nuke, and probably more components rely on the existing logic (e.g. comparisons for DCC versions) and they break because it doesn't always comply with [PEP 440](https://peps.python.org/pep-0440/). 

We won't be able to easily move to packaging. I'd suggest to keep using `LooseVersion` from the legacy distutils and keep suppressing the deprecation. Or fork the class code ourselves and strip out the deprecation.